### PR TITLE
Use `Balance` instead of `u128` as type for Balancer weights

### LINKED ISSUE #1213: [Swaps] Use a specific type for CPMM pool weights
Replace the hard-coded `u128` with a user-configurable type, either `Balance` or `Weight: From<Balacne> + Into<Balance>`. Probably the former.

### LINKED ISSUE #1216: [Primitives] Add `Weight` type to `SwapsApi`
Same as #1213, just for the `SwapsApi` trait.

### PR DESCRIPTION
**Pull request by maltekliemann**
### What does it do?

We replace the hardcoded `u128` for Balancer weights with the chain's balance type. We also replace a (benign) panicker in the CPMM math. Fixes #1213. Fixes #1216.

### CODE REVIEW

File: zrml/swaps/src/fixed.rs [hunk 1]
```diff
@@ -53,7 +55,7 @@ pub fn bsub_sign(a: u128, b: u128) -> Result<(u128, bool), DispatchError> {
 }
 
 pub fn bpowi(a: u128, n: u128) -> Result<u128, DispatchError> {
-    let mut z = if n % 2 != 0 { a } else { BASE };
```
Comment (Chralt98) [inline zrml/swaps/src/fixed.rs:56]: Oh, severe that we used `%` before 😰. But good reminder that `number % 0` produces a panic.

Comment (Chralt98) [reply zrml/swaps/src/fixed.rs:56]: I wonder, if we have other places using the `%` operator.. In court we have one for sure, but no zero division is possible there..

**Review by Chralt98 — APPROVED** [review]

File: zrml/swaps/src/fixed.rs (hunk 1 above)
Comment (Chralt98) [reply zrml/swaps/src/fixed.rs:56]: Could you update the court pallet regarding this change too in this PR?

Comment (maltekliemann) [reply zrml/swaps/src/fixed.rs]: #1254; reverted changes from this PR

**Review by sea212 — APPROVED** [review]

**Review by Chralt98 — APPROVED** [review]

**Pull request merged**